src/logsWebview.ts

```diff
@@ -0,0 +1,172 @@
+                var currentNonce = renderNonce;
+
+                var regexp = document.getElementById('regexp').value;
+                var mode = document.getElementById('mode').value;
+                if (lastMode == mode && lastRegexp == regexp) {
+                    return;
+                }
+                lastRegexp = regexp;
+                lastMode = mode;
+                if (regexp.length > 0) {
+                    var regex = new RegExp(regexp);
+                    switch (mode) {
+                        case 'all':
+                            content = orig;
+                            break;
+                        case 'include':
+                            content = orig.filter((line) => regex.test(line));
+                            break;
+                        case 'exclude':
+                            content = orig.filter((line) => !regex.test(line));
+                            break;
+                        case 'before':
+                            content = [];
+                            for (const line of orig) {
+                                if (regex.test(line)) {
+                                    break;
+                                }
+                                content.push(line);
+                            }
+                            break;
+                        case 'after':
+                            const i = orig.findIndex((line) => {
+                                return regex.test(line)
+                            });
+                            content = orig.slice(i+1);
+                            break;
+                    }
+                } else {
+                    content = orig;
+                }
+
+                var elt = document.getElementById('content');
+                elt.textContent = '';
+
+                // This is probably seems more complicated than necessary.
+                // However, rendering large blocks of text are _slow_ and kill the UI thread.
+                // So we split it up into manageable chunks to keep the UX lively.
+                // Of course the trouble is then we could interleave multiple different filters.
+                // So we use the random nonce to detect and pre-empt previous renders.
+                var ix = 0;
+                const step = 1000;
+                var fn = () => {
+                    if (renderNonce != currentNonce) {
+                        return;
+                    }
+                    if (ix >= content.length) {
+                        return;
+                    }
+                    var end = Math.min(content.length, ix + step);
+                    elt.appendChild(document.createTextNode(content.slice(ix, end).join('\\n')));
+                    ix += step;
+                    setTimeout(fn, 0);
+                }
+                fn();
+              };
+              eval();
+            </script>
+            </body>
+        </html>`;
+    }
+}
```

Comment:
wrong indentation level

Reply:
there is still a wrong indentation level. The indentation is correct upstream https://github.com/Azure/vscode-kubernetes-tools/blob/a7d45e92e39036256c610cec9cb0245c96453fb2/src/components/logs/logsWebview.ts#L161

Reply:
this is fine now (I should have put my comment on the line above so that it would have marked my comments as outdated automatically)